### PR TITLE
:warning: upgrade kubebuilder module from v2 to v3

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
upgrade kubebuilder module from v2 to v3

**Motivation**
https://github.com/kubernetes-sigs/kubebuilder/issues/1798

### CODE REVIEW

File: generate_testdata.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ source common.sh
 
 build_kb() {
   header_text "Building kubebuilder"
-  go build -o ./bin/kubebuilder ./cmd
+  go build -o ./bin/kubebuilder sigs.k8s.io/kubebuilder/v3/cmd
```
Comment (estroz) [inline generate_testdata.sh:21]: I don't think this is necessary

**Review by estroz — APPROVED** [review]
One question. Also @Adirio this doesn't affect the v2.3.2 release correct?

/approve

**Comment by camilamacedo86** [comment]
Hi @estroz, 

> One question. Also @Adirio this doesn't affect the v2.3.2 release correct?

How it would affect the  v2.3.2 release? It has its branch https://github.com/kubernetes-sigs/kubebuilder/tree/release-2 if we need to do a patch release.

File: generate_testdata.sh (hunk 1 above)
Comment (Adirio) [reply generate_testdata.sh:21]: This is definetely not needed and the `./cmd` will be valid forever.

**Comment by Adirio** [comment]
> Also @Adirio this doesn't affect the v2.3.2 release correct?

It shouldn't, its on a separate branch. We added v2 and it didnt affect v2.3.1 because it was already on a separate branch.

**Comment by Adirio** [comment]
My issue with this PR is that it will break any pending PR and right now there are several. It needs to be done but I think we need to time it a bit better.

**Comment by camilamacedo86** [comment]
Hi @Adirio, 

 > My issue with this PR is that it will break any pending PR and right now there are several. It needs to be done but I think we need to time it a bit better.

It is life. We should be done it already before the first nightly alpha release for v3. We need to move forward here to prepare a stable one. we cannot wait until have no longer open prs because it never will happen. So, It is time.

**Comment by Adirio** [comment]
The changes done here are correct except for the mentioned one above. Haven't checked if there ia any v2 remaining though.

File: generate_testdata.sh (hunk 1 above)
Comment (camilamacedo86) [reply generate_testdata.sh]: Solved.

**Pull request merged**